pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.

```

Comment:
I think it would be good to continue having "this parameter is trainable" as default, no?

Reply:
This is something that came up in the Variable refactor ADR. @co9olguy preferred if the wrapped NumPy library matched PyTorch, and was not trainable by default.

In the end, we decided to keep it trainable as default, to allow demos such as the qubit rotation one to continue to work. However, we marked this as something we wanted to revisit.